Entities/Scene.cpp

```diff
@@ -442,9 +442,8 @@ void Scene::Clear()
     m_AutoDesigned = true;
     m_TotalInvestment = 0;
     m_pTerrain = 0;
-   
-    for (int i = 0; i < sc_pathfinderCount; ++i) {
-        m_pPathFinders[i] = nullptr;
+    for (auto& pathFinder : m_pPathFinders) {
```

Comment:
I hate nitpicking like this, especially knowing you're not a fan of the symbol on the right, but you have a `&` on the left here.

Also, we tend to avoid using auto except when the type makes things less clear (e.g. longass iterators and such). Just to keep it more obvious on inspection what a thing is, versus hiding it away with auto. Applies below as well.